packages/compartment-mapper/src/bundle.js

```diff
@@ -271,22 +283,45 @@ ${__syncModuleProgram__},
   const cells = [
 ${''.concat(
   ...modules.map(
-    ({ record: { __fixedExportMap__, __liveExportMap__ } }) => `\
+    ({
+      record: { __fixedExportMap__, __liveExportMap__, __reexportMap__ },
+    }) => `\
     {
-${exportsCellRecord(__fixedExportMap__)}${exportsCellRecord(__liveExportMap__)}\
+${exportsCellRecord(__fixedExportMap__)}${exportsCellRecord(
+      __liveExportMap__,
+    )}${exportsCellRecord(adaptReexport(__reexportMap__))}\
     },
 `,
   ),
 )}\
   ];
 
 ${''.concat(
-  ...modules.flatMap(({ index, indexedImports, record: { reexports } }) =>
-    reexports.map(
-      importSpecifier => `\
+  ...modules.flatMap(
+    ({ index, indexedImports, record: { reexports, __reexportMap__ } }) => {
+      const mappings = reexports.map(
+        importSpecifier => `\
   Object.defineProperties(cells[${index}], Object.getOwnPropertyDescriptors(cells[${indexedImports[importSpecifier]}]));
 `,
-    ),
+      );
+      // Create references for export name as newname
+      const namedReexportsToProcess = Object.entries(__reexportMap__);
+      if (namedReexportsToProcess.length > 0) {
+        mappings.push(`
+  Object.defineProperties(cells[${index}], {${namedReexportsToProcess.map(
+          ([specifier, renames]) => {
+            return renames.map(
+              ([localName, exportedName]) =>
+                `${q(exportedName)}: { value: cells[${
+                  indexedImports[specifier]
+                }][${q(localName)}] }`,
+            );
+          },
+        )} });
+          `);
+      }
```

Comment:
I'm considering printing the 5 lines of closing braces as a poster or something.   
Spent non-trivial time matching all of them :sweat_smile:

Reply:
Code generation is fun. We need more Lisp macros.